source/renderer/app/components/widgets/forms/InlineEditingInput.js

```diff
@@ -76,122 +97,241 @@ export default class InlineEditingInput extends Component<Props, State> {
+    let error;
+    if (inputField.error) error = inputField.error;
+    else if (!hasChanged) error = errorMessage;
+
+    const showEditButton =
+      !isActive && !isLoading && !hasChanged && label.length && !readOnly;
+    const showFocusButtons =
+      !isLoading && !disabled && !readOnly && (isActive || hasChanged);
+    const showLoadingButton = isLoading;
+
     const componentStyles = classnames([
       className,
       styles.component,
       isActive ? null : styles.inactive,
+      readOnly ? styles.readOnly : null,
+      isLoading ? styles.isLoading : null,
+      showEditButton || showLoadingButton ? styles.twoButtons : null,
+      showFocusButtons ? styles.twoButtons : null,
     ]);
     const inputStyles = classnames([
       successfullyUpdated ? 'input_animateSuccess' : null,
       isActive ? null : 'input_cursorPointer',
     ]);
-
-    if (isActive) successfullyUpdated = false;
+    const buttonsWrapperStyles = classnames([
+      styles.buttonsWrapper,
+      readOnly ? styles.readOnly : null,
+    ]);
+    const editButtonStyles = classnames([styles.button, styles.editButton]);
+    const cancelButtonStyles = classnames([styles.button, styles.cancelButton]);
+    const okButtonStyles = classnames([styles.button, styles.okButton]);
+    const submittingButtonStyles = classnames([
+      styles.button,
+      styles.submittingButton,
+    ]);
 
     return (
-      <div
-        className={componentStyles}
-        onBlur={this.onBlur}
-        onMouseUp={this.onFocus}
-        role="presentation"
-        aria-hidden
-      >
+      <div className={componentStyles}>
         <Input
+          {...inputField.bind()}
           className={inputStyles}
+          placeholder={placeholder || ''}
           themeOverrides={styles}
           type="text"
           maxLength={maxLength}
-          label={inputFieldLabel}
-          value={inputField.value}
-          onChange={inputField.onChange}
-          onFocus={inputField.onFocus}
-          onBlur={inputField.onBlur}
+          label={label}
+          onFocus={this.onFocus}
+          onBlur={this.onBlur}
+          onChange={this.onChange}
           onKeyDown={(event) => this.handleInputKeyDown(event)}
-          error={isActive || inputBlocked ? inputField.error : null}
-          disabled={!isActive}
+          error={error}
+          disabled={disabled}
+          readOnly={readOnly}
+          showErrorState={!!error}
           ref={(input) => {
-            this.inputField = input;
+            if (!this.inputElement) {
+              this.inputElement = get(input, 'inputElement.current');
+            }
           }}
-          skin={InputSkin}
         />
 
-        {isActive && (
-          <button className={styles.button} onMouseDown={this.onCancel}>
-            {intl.formatMessage(messages.cancel)}
-          </button>
-        )}
+        <div
+          className={buttonsWrapperStyles}
+          onMouseDown={(event: KeyboardEvent) => {
```

Comment:
Same comment as above regarding function definition in `render` function.

Reply:
Done